code/modules/projectiles/guns/ballistic/rifle.dm

```diff
@@ -59,6 +61,30 @@ obj/item/gun/ballistic/rifle/attackby(obj/item/A, mob/user, params)
 	knife_x_offset = 27
 	knife_y_offset = 13
 
+/obj/item/gun/ballistic/rifle/boltaction/attackby(obj/item/A, mob/user, params)
+	..()
+	if(istype(A, /obj/item/circular_saw) || istype(A, /obj/item/gun/energy/plasmacutter))
```

Comment:
Still a bad way to check for what items can cut a gun.

Reply:
They're saws, that's what they do